src/file/functions.rs

```diff
@@ -281,30 +282,30 @@ pub fn copy_from_image(file_copy_params: &[FileCopyFromParams], image_file: &Pat
     for param in file_copy_params.iter() {
         let mut partition_file = working_dir.clone();
         let mut tmp_out_file = working_dir.clone();
-        let working_dir = working_dir.to_str().unwrap();
         let partition_info = get_partition_info(image_file, &param.partition)?;
         let in_file = param.in_file.to_str().unwrap();
 
         partition_file.push(Path::new(&format!("{}.img", partition_info.num)));
         let partition_file = partition_file.to_str().unwrap();
 
         read_partition(image_file, partition_file, &partition_info)?;
+        tmp_out_file.push(format!(
```

Comment:
@JanZachmann  Why did we decide to use a `tmp_out_file`? thinking about it. I'm currently not getting why we do need it.

Reply:
i tested it: in "check_file_copy_dos_partition" mcopy gets stuck, if the target file is not residing in the workingdir.
we need it when copying from "boot" partition, but not for the other partitions, where we use e2cp.